v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/SeparateShadowTableDatabaseIT.java

```diff
@@ -0,0 +1,309 @@
+  private static final String SPANNER_DDL_RESOURCE =
+      "SeparateShadowTableDatabaseIT/spanner-schema.sql";
+
+  private static HashSet<SeparateShadowTableDatabaseIT> testInstances = new HashSet<>();
+  private static PipelineLauncher.LaunchInfo jobInfo;
+
+  public static PubsubResourceManager pubsubResourceManager;
+  public static SpannerResourceManager spannerResourceManager;
+  public static SpannerResourceManager shadowSpannerResourceManager;
+
+  public SpannerResourceManager setUpShadowSpannerResourceManager() {
+    SpannerResourceManager sp =
+        SpannerResourceManager.builder("shadow_" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build();
+    sp.ensureUsableAndCreateResources();
+    return sp;
+  }
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException {
+    // Prevent cleaning up of dataflow job after a test method is executed.
+    skipBaseCleanup = true;
+    synchronized (SeparateShadowTableDatabaseIT.class) {
+      testInstances.add(this);
+      if (jobInfo == null) {
+        spannerResourceManager = setUpSpannerResourceManager();
+        shadowSpannerResourceManager = setUpShadowSpannerResourceManager();
+        pubsubResourceManager = setUpPubSubResourceManager();
+        createSpannerDDL(spannerResourceManager, SPANNER_DDL_RESOURCE);
+        jobInfo =
+            launchDataflowJob(
+                getClass().getSimpleName(),
+                null,
+                null,
+                "SeparateShadowTableDatabaseIT",
+                spannerResourceManager,
+                pubsubResourceManager,
+                new HashMap<>() {
+                  {
+                    put(
+                        "shadowTableSpannerInstanceId",
+                        shadowSpannerResourceManager.getInstanceId());
+                    put(
+                        "shadowTableSpannerDatabaseId",
+                        shadowSpannerResourceManager.getDatabaseId());
+                  }
+                },
+                null,
+                null);
+      }
+    }
+  }
+
+  /**
+   * Cleanup dataflow job and all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    for (SeparateShadowTableDatabaseIT instance : testInstances) {
+      instance.tearDownBase();
+    }
+    ResourceManagerUtils.cleanResources(
+        spannerResourceManager, shadowSpannerResourceManager, pubsubResourceManager);
+  }
+
+  /**
+   * This test works on a simple schema of 2 columns: id INT (PK), and val int. It tests INSERT,
+   * DELETE, UPDATE of val (non-pk) and update of id (PK, datastream generates an INSERT and DELETE
+   * event for PK updates) .
+   */
+  @Test
+  public void migrationTestSimpleTable() {
```

Comment:
is this there in the non-shadow table test as well ? It should ideally be a different test.

Reply:
What do you mean by non-shadow table test?

Both tests in this class use the same database, they just operate on separate tables. The nature of both the tests is the same, the only difference is one is a rudimentary schema and other is a more exhaustive on with all data types. 

Failure of both or one of these tests can help identify whether the issue lies with data type support or cross db txns in general.